### PR TITLE
Adding API to set BG scan parameters

### PR DESCRIPTION
**Pull request by AdhamAbozaeid**
Add new API pxSetBgScanParameters() to set BG scan parameters

Signed-off-by: Adham Abozaeid <adhsal@amazon.com>

Description
-----------
Added new API pxSetBgScanParameters() to set BG scan parameters

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ravibhagavandas — DISMISSED** [review]

File: libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h [hunk 1]
```diff
@@ -959,6 +959,19 @@ typedef struct
...
+     *
+     * @brief Sets the LE background scan interval and window in units of N*0.625 msec.
+     *
+     * @param[in] ucAdapterIf Adapter interface ID. Returned from BTRegisterBleAdapterCallback_t after calling pxRegisterBleApp.
+     * @param[in] ulScanInterval
+     * @param[in] ulScanWindow
+     * @return Returns eBTStatusSuccess on successful call.
+     */
+    BTStatus_t ( * pxSetBgScanParameters )( uint8_t ucAdapterIf,
+                                            uint32_t ulScanInterval,
```
Comment (gkwicker) [inline libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h:973]: It is preferable for time values to have a suffix that specifies the units, e.g. "ulTimeS", "ulTimeMs", "ulTimeSeconds", etc.  In this case the units are a little unusual - 5/8ms so this won't work.  However, the naming can at least specify that these are time values _and_ the units should also be listed in the doxygen parameter descriptions:

```c
  /**
    .
    .
     *
     * @param[in] ulScanIntervalTime - scan interval time in units of 0.625Ms
     * @param[in] ulScanWindowTime - scan window time in units of 0.625Ms
     .
     .
     .
     */
uint32_t ulScanIntervalTime,
uint32_t ulScanWindowTime
```

**Review by gkwicker — CHANGES_REQUESTED** [review]

File: libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h [hunk 2]
```diff
@@ -959,6 +959,19 @@ typedef struct
                                                bool bSetScanRsp,
                                                uint8_t * pucData,
                                                size_t xDataLen );
+
+    /**
+     *
+     * @brief Sets the LE background scan interval and window in units of N*0.625 msec.
+     *
+     * @param[in] ucAdapterIf Adapter interface ID. Returned from BTRegisterBleAdapterCallback_t after calling pxRegisterBleApp.
+     * @param[in] ulScanInterval scan interval time in units of 0.625Ms
```
Comment (gkwicker) [inline libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h:968]: Parameter names need to match the ones on lines 973/974.

**Review by gkwicker — CHANGES_REQUESTED** [review]
Needs a quick doxygen fix and then it is good to go

File: libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h (hunk 2 above)
Comment (AdhamAbozaeid) [reply libraries/abstractions/ble_hal/include/bt_hal_manager_adapter_ble.h]: Ops! Fixed that

**Review by gkwicker — APPROVED** [review]

**Review by ravibhagavandas — DISMISSED** [review]

**Review by ravibhagavandas — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

**Pull request merged**